### PR TITLE
SDK-189 [Python SDK] Schema 2.2 support 

### PR DESCRIPTION
**Pull request by RommeroInCountry**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- [SDK-189](https://incountry.atlassian.net/browse/SDK-189)

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version? What is the new version?

### CODE REVIEW

File: incountry/storage.py [hunk 1]
```diff
@@ -154,40 +252,100 @@ def find(
...
+        record_key_hashed = get_salted_hash(self.normalize_key(record_key), self.env_id)
+        self.http_client.delete(country=country, record_key=record_key_hashed)
         return {"success": True}
 
     @validate_encryption_enabled
     @validate_model(Country)
     @validate_model(FindFilter)
-    def migrate(self, country: str, limit: int = None) -> Dict[str, int]:
+    def migrate(self, country: str, limit: int = FIND_LIMIT) -> Dict[str, int]:
+        """Mirgrates records in InCountry storage to the latest encryption key.
```
Comment (artemgolozubov) [inline incountry/storage.py:326]: Pls fix typo `migrates`

File: tests/integration/test_methods.py [hunk 2]
```diff
@@ -147,131 +96,137 @@ def test_batch_write_records(
...
-@pytest.mark.parametrize("key", ["key2", "key3", "profile_key"])
+    keys_to_search = filter(lambda key: key not in ["body", "precommit_body"], get_valid_test_record().keys())
+    for key in keys_to_search:
+        key_value = choice(expected_records)[key]
+        if normalize_keys and isinstance(key_value, str):
+            key_value.should.equal(key_value.upper())
+        elif normalize_keys:
+            continue
+
+        print(f"Find by {key}{' with normalization' if normalize_keys else ''}.")
```
Comment (artemgolozubov) [inline tests/integration/test_methods.py:113]: Do we need this `print()` here?

Comment (artemgolozubov) [reply tests/integration/test_methods.py:113]: If we don't - please remove all `print()` calls below

Comment (RommeroInCountry) [reply tests/integration/test_methods.py:113]: to make testing logs more clear - previously  keys were a part of parametrized params which made tests really slow as test data was created each time. 
I'll probably update int tests in one of the future releases - but for now - that'll do I think

**Review by igor-alekseev-dev — APPROVED** [review]

**Review by artemgolozubov — APPROVED** [review]
LGTM

**Review by VladimirLevitskiy — APPROVED** [review]
LGTM

**Review by andrey-semakin — APPROVED** [review]
👍

**Pull request merged**